lib/sles4sap/sdaf_deployment_library.pm

```diff
@@ -827,6 +828,48 @@ sub get_sdaf_deployment_command {
     return $cmd;
 }
 
+=head2 git_clone
+
+    git_clone('https://github.com/myrepo/tree/main' [, branch=>'development', quiet=>'1', skip_ssl_verification=>'true',
+        output_log_file=>'git_clone.log']);
+
+B<repository>: Git repository url. Mandatory argument.
+
+B<branch>: Clone specific branch. Default: not defined
+
+B<quiet>: Minimize output verbosity. Default: false
+
+B<skip_ssl_verification>: Enable or disable SSL related error. Useful in case of self signed certificates. Default: false
+
+B<output_log_file>: Log output into a file.
+
+Generic wrapper around C<git clone> command. Supports basic set of switches and output logging.
+
+=cut
+
+sub git_clone {
+    my ($repository, %args) = @_;
+    croak 'Missing mandatory argument "repository"' unless $repository;
+
+    # Base command
+    my $git_cmd = 'git clone';
+
+    # Skip SSL verification
+    $git_cmd =~ s/git clone/git -c http.sslVerify=false clone/ if $args{skip_ssl_verification};
+
+    # Checkout branch
+    $git_cmd = "$git_cmd -b $args{branch}" if $args{branch};
+
+    # Append repository
+    $git_cmd = "$git_cmd $repository";
+
+    # Enable logging
+    $git_cmd = "set -o pipefail; $git_cmd 2>&1 | tee $args{output_log_file}" if $args{output_log_file};
+
+    record_info('git clone', "Cloning repository: $repository\nCMD: $git_cmd");
+    assert_script_run($git_cmd);
+}
+
```

Comment:
how about moving this to its on namespace? `Utils::Git` (`lib/Utils/Git.pm`)

Reply:
Umm... technically I could. 
I am scared to have the code on such a visible place though :D

But yeah... I'll move it there. B back once done.

Reply:
OK. I moved it to the new namespace as you suggested :)